### ISSUE TITLE
npm should not download or gunzip fsevents on a non-supported system

### ISSUE THREAD

**Issue by billymcintosh** [npm/npm#19747]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ x ] Other (_see below for feature requests_):

#### What's going wrong?
npm `5.6.0` **always** extracts **fsevents**  on a Windows based installation.
It's wasting time & cpu cycles - `fsevents` is supported on OSX only.    There should be no need to download its binaries nevermind gunzip them. 

It's extremely frustrating when doing 10+ `npm installs` per day.   

#### How can the CLI team reproduce the problem?
npm install something that contains fsevents.

### supporting information:
 - `npm -v` prints: 5.6.0
 - `node -v` prints: v8.9.4
 - `npm config get registry` prints :https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Windows 10
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by billymcintosh** [comment]
```
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64
```
`chokidar` _could_ be to blame...

```
        "chokidar": {
          "version": "1.5.2",
          ...snip...
          "requires": {
            ...snip...
            "fsevents": "1.1.3",
            ...snip...
          }
        },
```

**Comment by egarson** [comment]
Agreed, minor annoyance also here in Linux-land - the above resonates with me - at best it's annoying and at worst it's confusing to be told of "Unsupported platform for fsevents" on each npm interaction....

**Comment by Manningham** [comment]
I am seeing this issue too, and is affecting my docker workflow.  

If you do `npm info chokidar` you'll see that `fsevents` is both an optional dependency and a dependency... chokidar team is assigning blame to npm https://github.com/paulmillr/chokidar/issues/568

My current workaround is to manually remove `fsevents` from the package-lock.json file, which is obviously not a good thing to do, and then do my docker build.  May go back to yarn if they are handling optional dependencies better, haven't tried yet.
